app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart

```diff
@@ -0,0 +1,352 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show jsonEncode;
+
+import 'package:appengine/appengine.dart';
+import 'package:cocoon_service/src/request_handling/exceptions.dart';
+import 'package:graphql/client.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../foundation/providers.dart';
+import '../foundation/typedefs.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+
+const String _labeledPullRequestsWithReviewsQuery = r'''
+query LabeledPullRequestsWithReviews($sOwner: String!, $sName: String!, $sLabelName: String!) {
+  repository(owner: $sOwner, name: $sName) {
+    labels(first: 1, query: $sLabelName) {
+      nodes {
+        id
+        pullRequests(first: 100, states: OPEN) {
+          nodes {
+            id
+            number
+            mergeable
+            commits(last:1) {
+              nodes {
+                commit {
+                  abbreviatedOid
+                  oid
+                  status {
+                    state
+                  }
+                }
+              }
+            }
+            reviews(first: 100, states: [CHANGES_REQUESTED, APPROVED]) {
+              nodes {
+                state
+                author {
+                  login
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}''';
+
+const String _mergePullRequestMutation = r'''
+mutation MergePR($id: ID!, $oid: GitObjectID!) {
+  mergePullRequest(input: {
+    pullRequestId: $id,
+    expectedHeadOid: $oid,
+    mergeMethod: SQUASH,
+    commitBody: ""
```

Comment:
Can we by default copy the PR description into the merge commit description?

Reply:
That's usually not what you want in the commit description - it's very rarely formatted as a proper commit message and just dirties the log.  Thsi should result in just the PR title as the commit message.

Reply:
My habit is to always write a nice PR description that would be nice for merge commit message. When I merge manually, I'll copy it to the merge commit message.

It seems that others are doing similar things. Take the last 3 PRs as examples.
- https://github.com/flutter/flutter/pull/44151 has the related issue in the PR description.
- https://github.com/flutter/flutter/pull/44150 has the corresponding engine commits.
- https://github.com/flutter/flutter/pull/44149 has relevant information until the "checklist" section.

I think if auto-merged commits have an empty body, it will be very hard to check the `git log` to figure out what the commit is for (e.g., what issue does it fix).

Reply:
The first two are fine, but the third one is not something I'd want to see in a git log - and it's the only one of the three using our actual PR template :\

Reply:
I was using that template quite often until recently that I figured out that most people don't follow it anyway. In case that someone is using the template, it should be relatively simple to remove the "checklist" section automatically?